teamcity/resource_build_config.go

```diff
@@ -347,8 +347,24 @@ func resourceBuildConfigUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	if v, ok := d.GetOk("vcs_root"); ok {
+	if d.HasChange("vcs_root") {
+		oldVCSEntries, err := client.BuildTypes.GetVcsRootEntries(dt.ID)
+
+		if err != nil {
+			return err
+		}
+
+		for _, old := range oldVCSEntries.Items {
+			err := client.BuildTypes.DetachVcsRootEntry(dt.ID, old.ID)
+
+			if err != nil {
+				return err
+			}
+		}
+
+		v := d.Get("vcs_root")
```

Comment:
Should this be `GetOk` instead, with a check to only do lines 366-376 if `ok`? I'm thinking of the case where `HasChange` was true because all VCS roots were removed

Reply:
I've tested w/ removing all VCS roots and it's fine. I did it this way to match what the `"step"` version does, but I can change it to use the OK variant

Reply:
Just to confirm, does that mean that `v.(*schema.Set).List()` returns an empty map, even if `v` is nil?